experimental/checkpoint_test.go

```diff
@@ -0,0 +1,121 @@
+package experimental_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/experimental"
+	"github.com/tetratelabs/wazero/internal/testing/require"
+)
+
+func TestSnapshotNestedWasmInvocation(t *testing.T) {
```

Comment:
could you move this to internal/integration_test/adhoc.go? That's where we ensure the consistency across engines.

Reply:
Sure I can try it. Two issues

- This reuses the same snapshot.wasm used for the example test. Should I just copy the wasm?
- It's highly experimental, is it ok to have tests in the core area?

Reply:
ok let's leave it as-is now!